cmp/grid/Grid.ts

```diff
@@ -367,20 +351,78 @@ class GridLocalModel extends HoistModel {
         };
     }
 
+    //----------------------
+    // Row Height Management
+    //----------------------
+    @computed
+    get calculatedRowHeight() {
+        const {model} = this,
+            AgGridCmp = AgGrid as any;
+        return max([
+            AgGridCmp.getRowHeightForSizingMode(model.sizingMode),
+            maxBy(model.getVisibleLeafColumns(), 'rowHeight')?.rowHeight
+        ]);
+    }
+
+    @computed
+    get calculatedGroupRowHeight() {
+        const {sizingMode, groupRowHeight} = this.model,
+            {groupDisplayType} = this.agOptions,
+            AgGridCmp = AgGrid as any;
+        return groupRowHeight ?? groupDisplayType === 'groupRows'
+            ? AgGridCmp.getGroupRowHeightForSizingMode(sizingMode)
+            : AgGridCmp.getRowHeightForSizingMode(sizingMode);
+    }
+
+    defaultGetRowHeight = ({node}) => {
+        return node.group ? this.calculatedGroupRowHeight : this.calculatedRowHeight;
+    };
+
     rowHeightReaction() {
-        const {model} = this;
         return {
-            track: () => [model.getVisibleLeafColumns(), model.sizingMode],
-            run: ([visibleCols, sizingMode]: [Column[], SizingMode]) => {
-                this.fixedRowHeight = max([
-                    (AgGrid as any).getRowHeightForSizingMode(sizingMode),
-                    maxBy(visibleCols, 'rowHeight')?.rowHeight
-                ]);
+            track: () => [
+                this.useScrollOptimization,
+                this.calculatedRowHeight,
+                this.calculatedGroupRowHeight
+            ],
+            run: () => {
+                const {agApi} = this.model;
+                if (!agApi) return;
```

Comment:
Are we checking for `!agApi` b/c this might run before we have it, or b/c it might run after it's been disposed somehow?

If the former I am wondering if we should add to track

Reply:
agree, that is the normal pattern, but in this case, I think that case where agApi is filled in later, the data reaction will ensure the scroll optimization is applied at least once.